src/main/java/org/assertj/core/api/RecursiveIterableComparisonAssert.java

```diff
@@ -0,0 +1,285 @@
+
+    if(sizeOf((Iterable<?>) actual) != sizeOf(expected))
+      throw new AssertionError("Sizes are not the same.Actual is of size "
+                                          + sizeOf((Iterable<?>) actual)
+                                          + ", Expected is of size " + sizeOf(expected));
+
+    boolean ignoreActualIterableOrder = recursiveComparisonConfiguration.getIgnoreActualIterableOrder();
+    // order can not be ignored
+    if(!ignoreActualIterableOrder)
+      iterableAssert.isEqualTo(expected);
+    else {
+      // order can be ignored
+      for(Object element : expected)
+        iterableAssert.contains(element);
+    }
+
+    return myself;
+  }
+
+  /**
+   * Asserts that the iterable under test (actual) is equal to the given array when their elements are compared field by field recursively
+   * (inherited fields are included in the comparison). If the comparison fails it will report all the differences found and which
+   * effective {@link RecursiveComparisonConfiguration} was used to help users understand the failure.
+   * <p>
+   * This comparison, i.e Iterable to Array, is not allowed by default. To enable it, call {@link #allowingArrayTypeForExpected() allowingArrayTypeForExpected}/
+   * <p>
+   * All the comparison configuration behavior is documented in {@link RecursiveIterableComparisonAssert#isEqualTo(Object) RecursiveComparisonAssert#isEqualTo}
+   * is valid also for the comparison made on the elements of actual and expected iterables.
+   *
+   * <pre><code class='java'>public class Person {
+   *     String name;
+   *     boolean hasPhd;
+   * }
+   *
+   * Person sheldon = new People("Sheldon Cooper", true);
+   * Person howard = new People("Howard Wolowitz", false);
+   *
+   * Person[] peopleArray = {sheldon, howard};
+   * List<Person> peopleList = Arrays.asList(peopleArray);
+   *
+   * // assertion fails because expected object can't be of type Array.
+   * assertThat(peopleList).usingRecursiveComparison()
+   *                       .isEqualTo(peopleArray);
+   *
+   * // assertion succeeds because array type is explicitly made allowed.
+   * assertThat(peopleList).usingRecursiveComparison()
+   *                       .allowingArrayTypeForExpected()
+   *                       .isEqualTo(peopleArray);</code></pre>
+   *
+   * @param values the objects that must be contained in {@code actual}.
+   * @return {@code this} assertion object.
+   * @throws AssertionError if the actual and the given objects are not deeply equal property/field by property/field.
+   * @throws IntrospectionError if one property/field to compare can not be found.
+   */
+  public SELF isEqualTo(Object... values) {
+    if(!recursiveComparisonConfiguration.getAllowArrayTypeForExpected())
+      throw new AssertionError("Type array is not acceptable when comparing iterables, try calling #allowingArrayTypeForExpected configuration.");
+    return isEqualTo(Arrays.asList(values));
```

Comment:
use AssertJ the more elegant `Lists.list` and static import it for optimal readability.